engine/src/main/java/org/terasology/entitySystem/entity/internal/EngineEntityPool.java

```diff
@@ -69,4 +71,20 @@
      * @return whether the entity has the component
      */
     boolean hasComponent(long entityId, Class<? extends Component> componentClass);
+
+    /**
+     * Remove the entity from the pool. This does not destroy the entity, it only removes the {@link EntityRef}
+     * and the {@link Component}s from this pool, so that the entity can be moved to a different pool. It does
+     * not invalidate the {@link EntityRef}.
+     *
+     * Returns an {@link Optional} {@link BaseEntityRef} if it was removed, ready to be put into another pool. If nothing
+     * was removed, return {@link Optional#empty()}.
+     *
+     * @param id the id of the entity to remove
+     * @return an optional {@link BaseEntityRef}, containing the removed entity
+     */
+    Optional<BaseEntityRef> remove(long id);
```

Comment:
Yes, that would be a problem. If this is called without caution, all of the components will be removed from the entity, but the entity will still stay registered and the ref could be re-created.

The method is intended for use only by `moveToPool()`, but I had to put it in the interface because the entity manager currently deals in terms of EngineEntityPools, rather than PojoEntityPools. This makes it more general and means that an alternative implementation of pool could be easily substituted, however it means that any methods like this have to be included in the interface.

I don't think that's a huge problem, but it's probably worth adding a note warning against improper use (unless you can think of a better way of doing it).